### PR TITLE
add sprinklr type converter per sbp_2474

### PR DESCRIPTION
**Pull request by eodgooch**
work to add sprinklr type converter per ticket sbp_2474

### CODE REVIEW

File: streams-contrib/streams-provider-sprinklr/pom.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            <artifactId>streams-pojo-extensions</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.streams</groupId>
+            <artifactId>streams-config</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.streams</groupId>
+            <artifactId>streams-util</artifactId>
+            <version>0.2.21-W2O-SNAPSHOT</version>
```
Comment (sieverssj) [inline streams-contrib/streams-provider-sprinklr/pom.xml]: This should be ${project.version}

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/processor/SprinklrTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                count++;
+            }
+
+            // maybe it's something we don't even know about yet
+            else {
+                LOGGER.info("{} processing unknown item class={}", STREAMS_ID, item.getClass());
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (sieverssj) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/processor/SprinklrTypeConverter.java]: Logging should be sufficient here.  You can do LOGGER.error("your message", e) and the whole exception, including the stack trace will be logged.

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+     */
+    public Activity convert(JsonNode item) {
+        Activity activity = new Activity();
+
+        try {
+            activity.setVerb("post");
+            activity.setId(formatId(activity.getVerb(), item.get("universalMessageId").asText()));
+            activity.setContent(item.has("message") ? item.get("message").asText() : "");
+            activity.setTitle(item.has("title") ? item.get("title").asText() : "");
+            activity.setPublished(item.has("createdTime") ? new DateTime(item.get("createdTime").asLong()) : new DateTime());
```
Comment (sieverssj) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java]: "createdTime" is the time the content was created on Twitter (for instance) or in the Sprinklr system?

**Comment by sieverssj** [comment]
I see SprinklrData.json but it doesn't appear to have anything in it.  Are you waiting for more information?

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    }
+
+    /**
+     * Set Activity Extension properties for location and language
+     * @param item JsonNode
+     * @param extensions a Map of String to Object
+     */
+    @SuppressWarnings("unchecked")
+    private void setActivityExtensions(JsonNode item, Map<String, Object> extensions) {
+        Map<String, Object> locationMap;
```
Comment (robdouglas) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java]: I haven't looked too deep into the data that we can get back from Sprinklr, but do posts have any information about things like favorites/likes count or rebroadcast counts?

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java (hunk 3 above)
Comment (eodgooch) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java]: I went with the content creation time.

Per the documentation:

- ‘SYSTEM_CREATED_TIME’ - system created time. 
- ‘CREATED_TIME’ - social network created time. 
- ‘MODIFIED_TIME’ - modified time.

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java (hunk 4 above)
Comment (eodgooch) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java]: Sprinklr returns different data for each social network. I have gone through the provided documentation and updated the mapping to include more details of the activity.

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/processor/SprinklrTypeConverter.java (hunk 2 above)
Comment (eodgooch) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/processor/SprinklrTypeConverter.java]: Sorry, bad copy and paste.

File: streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java (hunk 4 above)
Comment (eodgooch) [inline streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java]: This work is done. Are we still adding it in?

**Pull request merged**